### ISSUE TITLE
[urgent] Inputs to operation AddN_7 of type AddN must have the same size and shape.

### ISSUE THREAD

**Issue by JoshuaChou2018** [perslev/MultiPlanarUNet#10]
I used the v2.5.0 and choose the loss function "SparseGeneralizedDiceLoss" and metrics "sparse_categorical_accuracy" and tried both Adam and SGD

please help, thanks!

then the error gives:

Epoch 1/25
2020-04-03 09:36:51.274232: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2020-04-03 09:36:53.101431: W tensorflow/stream_executor/gpu/redzone_allocator.cc:312] Not found: ./bin/ptxas not found
Relying on driver to perform ptx compilation. This message will be only logged once.
2020-04-03 09:36:53.352288: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2020-04-03 09:36:57.993085: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Invalid argument: Inputs to operation AddN_7 of type AddN must have the same size and shape.  Input 0: [1,1] != input 2: [0,1]
         [[{{node AddN_7}}]]
         [[SGD/SGD/update_2_1/AssignAddVariableOp/_1607]]
2020-04-03 09:36:57.994718: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Invalid argument: Inputs to operation AddN_7 of type AddN must have the same size and shape.  Input 0: [1,1] != input 2: [0,1]
         [[{{node AddN_7}}]]
         [[GroupCrossDeviceControlEdges_0/SGD/SGD/update_1_1/Const/_1471]]
2020-04-03 09:36:57.997156: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Invalid argument: Inputs to operation AddN_7 of type AddN must have the same size and shape.  Input 0: [1,1] != input 2: [0,1]
         [[{{node AddN_7}}]]
         [[GroupCrossDeviceControlEdges_1/SGD/SGD/update_0/Const/_1407]]
2020-04-03 09:36:57.997288: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Invalid argument: Inputs to operation AddN_7 of type AddN must have the same size and shape.  Input 0: [1,1] != input 2: [0,1]
         [[{{node AddN_7}}]]
         [[GroupCrossDeviceControlEdges_0/SGD/SGD/update_0/Const/_1391]]
2020-04-03 09:36:58.002164: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Invalid argument: Inputs to operation AddN_7 of type AddN must have the same size and shape.  Input 0: [1,1] != input 2: [0,1]
         [[{{node AddN_7}}]]
         [[GroupCrossDeviceControlEdges_4/SGD/SGD/update_0/Const/_1443]]
2020-04-03 09:36:58.002986: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Invalid argument: Inputs to operation AddN_7 of type AddN must have the same size and shape.  Input 0: [1,1] != input 2: [0,1]
         [[{{node AddN_7}}]]
         [[SGD/SGD/group_deps/_1643]]
2020-04-03 09:36:58.003135: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Invalid argument: Inputs to operation AddN_7 of type AddN must have the same size and shape.  Input 0: [1,1] != input 2: [0,1]
         [[{{node AddN_7}}]]
         [[GroupCrossDeviceControlEdges_4/SGD/SGD/update_0/Const/_1571]]
2020-04-03 09:36:58.003238: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Invalid argument: Inputs to operation AddN_7 of type AddN must have the same size and shape.  Input 0: [1,1] != input 2: [0,1]
         [[{{node AddN_7}}]]
         [[GroupCrossDeviceControlEdges_6/SGD/SGD/update_0/Const/_1535]]
2020-04-03 09:36:58.003819: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Invalid argument: Inputs to operation AddN_7 of type AddN must have the same size and shape.  Input 0: [1,1] != input 2: [0,1]
         [[{{node AddN_7}}]]
   1/1250 [..............................] - ETA: 14:00:49
WARNING:tensorflow:Using MirroredStrategy eagerly has significant overhead currently. We will be working on improving this in the future, but for now please wrap `call_for_each_replica` or `experimental_run` or `experimental_run_v2` inside a tf.function to get the best performance.
WARNING:tensorflow:Using MirroredStrategy eagerly has significant overhead currently. We will be working on improving this in the future, but for now please wrap `call_for_each_replica` or `experimental_run` or `experimental_run_v2` inside a tf.function to get the best performance.
WARNING:tensorflow:Using MirroredStrategy eagerly has significant overhead currently. We will be working on improving this in the future, but for now please wrap `call_for_each_replica` or `experimental_run` or `experimental_run_v2` inside a tf.function to get the best performance.
WARNING:tensorflow:Using MirroredStrategy eagerly has significant overhead currently. We will be working on improving this in the future, but for now please wrap `call_for_each_replica` or `experimental_run` or `experimental_run_v2` inside a tf.function to get the best performance.
WARNING:tensorflow:Using MirroredStrategy eagerly has significant overhead currently. We will be working on improving this in the future, but for now please wrap `call_for_each_replica` or `experimental_run` or `experimental_run_v2` inside a tf.function to get the best performance.
